definitions/npm/openai_v4.x.x/flow_v0.142.x-/openai_v4.x.x.js

```diff
@@ -0,0 +1,2515 @@
+     */
+    create(
+      body: EmbeddingCreateParams,
+      options?: RequestOptions<>
+    ): APIPromise<typeof CreateEmbeddingResponse>;
+  }
+  declare var CreateEmbeddingResponse: typeof npm$namespace$CreateEmbeddingResponse;
+
+  declare var npm$namespace$CreateEmbeddingResponse: {|
+    Usage: Class<CreateEmbeddingResponse$Usage>,
+  |};
+
+  /**
+   * The usage information for the request.
+   */
+  declare interface CreateEmbeddingResponse$Usage {
+    /**
+     * The number of tokens used by the prompt.
+     */
+    prompt_tokens: number;
+
+    /**
+     * The total number of tokens used by the request.
+     */
+    total_tokens: number;
+  }
+  /**
+   * Represents an embedding vector returned by embedding endpoint.
+   */
+  declare interface Embedding {
+    /**
+     * The embedding vector, which is a list of floats. The length of vector depends on
+     * the model as listed in the
+     * [embedding guide](https://platform.openai.com/docs/guides/embeddings).
+     */
+    embedding: Array<number>;
+
+    /**
+     * The index of the embedding in the list of embeddings.
+     */
+    index: number;
+
+    /**
+     * The object type, which is always "embedding".
+     */
+    object: string;
+  }
+  declare interface EmbeddingCreateParams {
+    /**
+     * Input text to embed, encoded as a string or array of tokens. To embed multiple
+     * inputs in a single request, pass an array of strings or array of token arrays.
+     * Each input must not exceed the max input tokens for the model (8191 tokens for
+     * `text-embedding-ada-002`) and cannot be an empty string.
+     * [Example Python code](https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb)
+     * for counting tokens.
+     */
+    input: string | Array<string> | Array<number> | Array<Array<number>>;
+
+    /**
+     * ID of the model to use. You can use the
+     * [List models](https://platform.openai.com/docs/api-reference/models/list) API to
+     * see all of your available models, or see our
+     * [Model overview](https://platform.openai.com/docs/models/overview) for
+     * descriptions of them.
+     */
+    model: "text-embedding-ada-002";
+
+    /**
+     * A unique identifier representing your end-user, which can help OpenAI to monitor
+     * and detect abuse.
+     * [Learn more](https://platform.openai.com/docs/guides/safety-best-practices/end-user-ids).
+     */
+    user?: string;
+  }
+  declare class Edits extends APIResource {
+    /**
+     * Creates a new edit for the provided input, instruction, and parameters.
+     * @deprecated The Edits API is deprecated; please use Chat Completions instead.
+     *
+     * https://openai.com/blog/gpt-4-api-general-availability#deprecation-of-the-edits-api
+     */
+    create(body: EditCreateParams, options?: RequestOptions<>): APIPromise<typeof Edit>;
+  }
+  declare var Edit: typeof npm$namespace$Edit;
+
+  declare var npm$namespace$Edit: {|
```

Comment:
Just wondering is this meant to be a variable or a type is better?

Reply:
Thanks; fixed in 181b181a. I lost something because of identical namespace and class names in the typescript.